editor/src/components/canvas/controls/select-mode/padding-resize-control.tsx

```diff
@@ -0,0 +1,262 @@
+import React from 'react'
+import { CanvasVector, windowPoint } from '../../../../core/shared/math-utils'
+import { assertNever } from '../../../../core/shared/utils'
+import { Modifier } from '../../../../utils/modifiers'
+import { useColorTheme } from '../../../../uuiui'
+import { EditorDispatch } from '../../../editor/action-types'
+import { EditorStorePatched } from '../../../editor/store/editor-state'
+import { useEditorState, useRefEditorState } from '../../../editor/store/store-hook'
+import CanvasActions from '../../canvas-actions'
+import { controlForStrategyMemoized } from '../../canvas-strategies/canvas-strategy-types'
+import {
+  createInteractionViaMouse,
+  paddingResizeHandle,
+} from '../../canvas-strategies/interaction-state'
+import { CSSCursor, EdgePiece } from '../../canvas-types'
+import { windowToCanvasCoordinates } from '../../dom-lookup'
+import { simplePaddingFromMetadata } from '../../padding-utils'
+import { useBoundingBox } from '../bounding-box-hooks'
+import { CanvasOffsetWrapper } from '../canvas-offset-wrapper'
+import { isZeroSizedElement } from '../outline-utils'
+import { useMaybeHighlightElement } from './select-mode-hooks'
+
+export const paddingControlTestId = (edge: EdgePiece): string => `padding-control-${edge}`
+export const paddingControlHandleTestId = (edge: EdgePiece): string =>
+  `padding-control-handle-${edge}`
+
+type Orientation = 'vertical' | 'horizontal'
+
+interface ResizeContolProps {
+  edge: EdgePiece
+  hiddenByParent: boolean
+}
+
+const transformFromOrientation = (orientation: Orientation) => {
+  switch (orientation) {
+    case 'horizontal':
+      return '90deg'
+    case 'vertical':
+      return '0deg'
+    default:
+      assertNever(orientation)
+  }
+}
+
+type Timeout = ReturnType<typeof setTimeout>
+
+const PaddingResizeControlWidth = 4
+const PaddingResizeControlHeight = 24
+const PaddingResizeControlBorder = 1
+const PaddingResizeControlI = React.memo(
+  React.forwardRef<HTMLDivElement, ResizeContolProps>((props, ref) => {
+    const scale = useEditorState((store) => store.editor.canvas.scale, 'PaddingResizeControl scale')
+    const dispatch = useEditorState((store) => store.dispatch, 'PaddingResizeControl dispatch')
+    const canvasOffsetRef = useRefEditorState((store) => store.editor.canvas.roundedCanvasOffset)
+    const { maybeClearHighlightsOnHoverEnd } = useMaybeHighlightElement()
+
+    const colorTheme = useColorTheme()
+
+    const [hidden, setHidden] = React.useState<boolean>(true)
+    const [hoverStart, hoverEnd] = useHoverWithDelay(200, (h) => setHidden(!h))
```

Comment:
Shouldn't this just be calling `setHidden(false)`?

Reply:
Ok nopr, it seems I've misunderstood what the param for `useHoverWithDelay` was

Reply:
should I come up with a better name for `useHoverWithDelay`?